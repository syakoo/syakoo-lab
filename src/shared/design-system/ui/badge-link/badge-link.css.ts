import { style } from "@vanilla-extract/css";

import { theme } from "@/shared/design-system/theme.css";

export const styles = {
  container: style({
    // NOTE: ボーダーの幅を考慮して、paddingを調整する
    paddingBlock: `calc(${theme.space[50]} - 1px)`,
    paddingRight: `calc(${theme.space[100]} - 1px)`,
    paddingLeft: `calc(${theme.space[50]} - 1px)`,
    borderRadius: theme.radius[300],
    border: `1px solid ${theme.color.text.tertiary}`,

    ":hover": {
      backgroundColor: theme.color.background.secondary,
      borderColor: theme.color.text.secondary,
    },
  }),

  iconWrapper: style({
    width: theme.fontSize[400],
    height: theme.fontSize[400],
    padding: theme.space[50],
    borderRadius: theme.radius[1000],
    display: "flex",
  }),
  image: style({
    borderRadius: "100px",
    objectFit: "cover",
  }),
};
