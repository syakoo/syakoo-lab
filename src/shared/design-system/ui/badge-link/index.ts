export { BadgeLink } from "./badge-link";
