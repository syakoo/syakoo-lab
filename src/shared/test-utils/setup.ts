import { exampleMDXComponent } from "@/features/mdx/mocks/fixture-mdxcomponent";

import { loadEnv } from "./nextjs/env";

// NOTE: `@mdx-js/mdx` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
jest.mock("@mdx-js/mdx", () => {
  return {
    runSync: jest
      .fn()
      .mockImplementation(() => ({ default: exampleMDXComponent })),
  };
});

// NOTE: `mermaid` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
jest.mock("mermaid", () => ({
  initialize: () => {
    return;
  },
  run: () => {
    return;
  },
}));

loadEnv();

jest.spyOn(Math, "random").mockImplementation(() => 0.5866245602626035);
