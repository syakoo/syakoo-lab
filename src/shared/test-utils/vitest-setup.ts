import { vi } from "vitest";

import { exampleMDXComponent } from "@/features/mdx/mocks/fixture-mdxcomponent";

import { loadEnv } from "./nextjs/env";

// NOTE: Next.jsのImageコンポーネントをテスト用にモック
vi.mock("next/image", () => ({
  default: "img",
}));

// NOTE: `@mdx-js/mdx` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
vi.mock("@mdx-js/mdx", () => {
  return {
    runSync: vi
      .fn()
      .mockImplementation(() => ({ default: exampleMDXComponent })),
  };
});

// NOTE: `mermaid` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
vi.mock("mermaid", () => ({
  default: {
    initialize: () => {
      return;
    },
    run: () => {
      return;
    },
  },
  initialize: () => {
    return;
  },
  run: () => {
    return;
  },
}));

loadEnv();

vi.spyOn(Math, "random").mockImplementation(() => 0.5866245602626035);
