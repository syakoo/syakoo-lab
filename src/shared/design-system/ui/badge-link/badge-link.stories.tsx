import type { Meta, StoryObj } from "@storybook/nextjs";

import { BadgeLink } from "./badge-link";

const meta = {
  component: BadgeLink,
  parameters: {
    layout: "centered",
    docs: {
      description: {
        component:
          "https://www.figma.com/design/KM3QHHUgriWxt0RJ1MT5SB/syakoo-lab?node-id=543-1302&t=kVsF3rxs1WNK1866-4",
      },
    },
    testLevel: "snapshot",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof BadgeLink>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Sample: Story = {
  args: {
    href: "https://syakoo-lab.com",
    color: "rgb(48, 99, 212)",
    children: "Syakoo Lab",
  },
};
