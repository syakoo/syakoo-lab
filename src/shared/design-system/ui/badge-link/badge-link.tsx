import Image from "next/image";

import { Row } from "@/shared/design-system/layout";
import { Link, Span } from "@/shared/design-system/ui";

import { styles } from "./badge-link.css";

export type BadgeLinkProps = {
  href: string;
  imageUrl: string;
  children: string;
};

export const BadgeLink: React.FC<BadgeLinkProps> = ({
  href,
  imageUrl,
  children,
}) => {
  return (
    <Link href={href} noTransparent>
      <div className={styles.container}>
        <Row align="center" gap="50">
          <Image
            alt=""
            className={styles.image}
            height={24}
            src={imageUrl}
            width={24}
          />
          <Span color="primary" size="100">
            {children}
          </Span>
        </Row>
      </div>
    </Link>
  );
};
