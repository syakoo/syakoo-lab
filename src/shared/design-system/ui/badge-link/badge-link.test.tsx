import { testStories } from "@/shared/test-utils/test-stories";

import * as stories from "./badge-link.stories";

describe("BadgeLink", () => {
  testStories(stories);
});
