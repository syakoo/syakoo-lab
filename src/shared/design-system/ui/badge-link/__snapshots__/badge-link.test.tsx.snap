// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BadgeLink > スナップショット Sample 1`] = `
<body>
  <div>
    <a
      class="link_linkStyle__1vz8ftz0 link_linkStyle_noHovered_true__1vz8ftz3 link_linkStyle_noTransparent_true__1vz8ftz5"
      href="https://syakoo-lab.com"
    >
      <div
        class="badge-link_styles_container__jnr83m0"
      >
        <div
          class="flex_flexStyle__zqca9ti flex_flexStyle_align_center__zqca9ts flex_flexStyle_variants_align_center__zqca9t7 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_50__zqca9tx flex_flexStyle_variants_gap_50__zqca9tc flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
        >
          <div
            class="badge-link_styles_iconWrapper__jnr83m1"
            style="background-color: rgb(48, 99, 212);"
          >
            <svg
              color="var(--color-text-primary__1874wf7c)"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.19 8.688a4.5 4.5 0 0 1 1.242 7.244l-4.5 4.5a4.5 4.5 0 0 1-6.364-6.364l1.757-1.757m13.35-.622 1.757-1.757a4.5 4.5 0 0 0-6.364-6.364l-4.5 4.5a4.5 4.5 0 0 0 1.242 7.244"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
          <span
            class="text_textStyle__1djq151e text_textStyle_color_primary__1djq151f text_textStyle_variants_color_primary__1djq1510 text_textStyle_size_75__1djq151k text_textStyle_variants_size_75__1djq1515 text_textStyle_weight_normal__1djq151u"
          >
            Syakoo Lab
          </span>
        </div>
      </div>
    </a>
  </div>
</body>
`;
